apps/04TowersOfHanoi.js

```diff
@@ -19,24 +19,61 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack, endStack) {
   // Your code here
-
+  //putting popped index item in a variable so that it can be pushed onto endStack once it has been popped
+  var poppedItem = stacks[startStack].pop();
+  stacks[endStack].push(poppedItem);
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
   // Your code here
 
+  var start = stacks[startStack];
+  var end = stacks[endStack];
+
+  //getting the last element of the array, to be ready for comparison in later 'if' statements
+
+  var lastElementOfStartStack = start[start.length - 1];
+  var lastElementOfEndStack = end[end.length - 1];
+
+
+  //if the startStack doesnt have anything there, nothing can be moved from it
+  if (start.length === 0) {
```

Comment:
Nice early returns! You could probably bump these up a few lines to _before_ you even declare the lastElement variables